Src/FluentAssertions/Collections/GenericCollectionAssertions.cs

```diff
@@ -349,6 +349,124 @@ private bool IsValidProperty<TSelector>(Expression<Func<T, TSelector>> propertyE
             BeEquivalentTo(repeatedExpectation, forceStringOrderingConfig, because, becauseArgs);
         }
 
+        /// <summary>
+        /// Asserts that a collection contains exactly a given number of elements, which meet
+        /// the criteria provided by the element inspectors.
+        /// </summary>
+        /// <param name="elementInspectors">
+        /// The element inspectors, which inspect each element in turn. The
+        /// total number of element inspectors must exactly match the number of elements in the collection.
+        /// </param>
+        public void SatisfyRespectively(params Action<T>[] elementInspectors)
+        {
+            SatisfyRespectively(elementInspectors, string.Empty);
+        }
+
+        /// <summary>
+        /// Asserts that a collection contains exactly a given number of elements, which meet
+        /// the criteria provided by the element inspectors.
+        /// </summary>
+        /// <param name="expected">
+        /// The element inspectors, which inspect each element in turn. The
+        /// total number of element inspectors must exactly match the number of elements in the collection.
+        /// </param>
+        /// <param name="because">
+        /// An optional formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the
+        /// assertion is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public void SatisfyRespectively(IEnumerable<Action<T>> expected, string because = "", params object[] becauseArgs)
+        {
+            if (expected is null)
+            {
+                throw new ArgumentNullException(nameof(expected), "Cannot verify against a <null> collection of inspectors");
+            }
+
+            ICollection<Action<T>> elementInspectors = expected.ConvertOrCastToCollection();
+            if (!elementInspectors.Any())
+            {
+                throw new ArgumentException("Cannot verify against an empty collection of inspectors", nameof(expected));
+            }
+
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .WithExpectation("Expected {context:collection} to satisfy all inspectors{reason}, ")
+                .ForCondition(!(Subject is null))
+                .FailWith("but collection is <null>.")
+                .Then
+                .ForCondition(Subject.Any())
+                .FailWith("but collection is empty.")
+                .Then
+                .ClearExpectation();
+
+            int elementsCount = Subject.Count();
+            int inspectorsCount = elementInspectors.Count;
+            if (elementsCount != inspectorsCount)
+            {
+                if (inspectorsCount == 1)
```

Comment:
I would simplify those two situations into a single generic statement:

`Expected {context:collection} to contain exactly {0} items, but it contained {1}  items"`

If the count doesn't match, it does not matter what kind of inspectors you used.

Reply:
Updated as suggested.